src/category/category.service.ts

```diff
@@ -0,0 +1,63 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { FindOptionsWhere } from 'typeorm';
+
+import { categoryMessages } from '@common/constants/messages';
+import { CreateCategoryDto } from './dto/create-category.dto';
+import { UpdateCategoryDto } from './dto/update-category.dto';
+import { CategoryRepository } from './repositories/category.repository';
+import { Category } from './entities/category.entity';
+
+@Injectable()
+export class CategoryService {
+  constructor(private readonly categoryRepository: CategoryRepository) {}
+
+  async findAll() {
+    const categories = await this.categoryRepository.find();
+    return categories;
+  }
+
+  async findOne(id: string) {
+    const category = await this.categoryRepository.findOne({
+      where: { id },
+    });
+    if (!category)
+      throw new BadRequestException(categoryMessages.CATEGORY_NOT_FOUND);
```

Comment:
I think that this should be a NotFoundException

Reply:
right